spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/SpanSubscriber.java

```diff
@@ -81,7 +88,7 @@ public void onSubscribe(Subscription subscription) {
 			log.trace("On subscribe");
 		}
 		this.s = subscription;
-		try (Tracer.SpanInScope ws = this.tracer.withSpanInScope(this.span)) {
+		try (Scope ws = this.currentTraceContext.maybeScope(this.span.context())) {
```

Comment:
if the span was already in scope, this would have synced MDC etc again for no reason